website/docs/r/cos_bucket_lifecycle_configuration.markdown

```diff
@@ -0,0 +1,324 @@
+  bucket_crn = ibm_cos_bucket.cos_bucket.crn
+  bucket_location = ibm_cos_bucket.cos_bucket.region_location
+  lifecycle_rule {
+    noncurrent_version_expiration{
+			   noncurrent_days = "1
+		}
+    filter {
+      prefix = ""
+    }  
+    rule_id = "id"
+    status = "Enabled"
+  
+  }
+}
+
+```
+# Adding lifecycle configuration with multiple rules.
+
+```terraform
+
+
+resource "ibm_cos_bucket" "cos_bucket" {
+  bucket_name           = var.bucket_name
+  resource_instance_id  = ibm_resource_instance.cos_instance.id
+  region_location       = var.regional_loc
+  storage_class         = var.standard_storage_class
+
+}
+resource "ibm_cos_bucket_lifecycle_configuration"  "lifecycle" {
+  bucket_crn = ibm_cos_bucket.cos_bucket.crn
+  bucket_location = ibm_cos_bucket.cos_bucket.region_location
+  lifecycle_rule {
+    expiration{
+      days = 1
+    }
+    filter {
+      prefix = "foo"
+    }  
+    rule_id = "id"
+    status = "Enabled"
+  }
+    lifecycle_rule {
+    expiration{
+      days = 2
+    }
+    filter {
+      prefix = "bar"
+    }  
+    rule_id = "id2"
+    status = "Enabled"
+  }
+}
+
+```
+
+# Switching from legacy lifecycle rules to ibm_cos_bucket_lifecycle_configuration :
+
+
+**Note:**
+If you use legacy `expire_rule` , `archive_rule` , `noncurrent_version_expiration`, `abort_incomplete_multipart_upload_days`  on an ibm_cos_bucket, Terraform will assume management over the full set of Lifecycle rules for the bucket, treating additional Lifecycle rules as drift. For this reason, legacy rules cannot be mixed with the external ibm_cos_bucket_lifecycle_configuration resource for a given cos bucket.Users that want to continue using the legacy `expire_rule` , `archive_rule` , `noncurrent_version_expiration`, `abort_incomplete_multipart_upload_days`. 
+
+In case you want to switch from the legacy lifecycle rules to the new resource for an existing cos bucket with existing legacy lifecycle rules , please follow the steps below
+
+## Example usage
+For example if there is a cos bucket existing in the `.tfstate` file with `expire_rule` applied.
+```terraform
+
+# Existing Cos bucket configuration 
+
+resource "ibm_cos_bucket" "cos_bucket" {
+  bucket_name           = var.bucket_name
+  resource_instance_id  = ibm_resource_instance.cos_instance.id
+  region_location       = var.regional_loc
+  storage_class         = var.standard_storage_class
+  expire_rule {
+    rule_id = "id"
+    enable  = true
+    days    = 45
+    prefix  = "foo"
+  }
+
+}
+
+```
+Step 1 : Add new `ibm_cos_bucket_lifecycle_configuration` with same configuration as exisitng.
+
+```terraform
+
+resource "ibm_cos_bucket" "cos_bucket" {
+  bucket_name           = var.bucket_name
+  resource_instance_id  = ibm_resource_instance.cos_instance.id
+  region_location       = var.regional_loc
+  storage_class         = var.standard_storage_class
+  expire_rule {
+    rule_id = "id"
+    enable  = true
+    days    = 45
+    prefix  = "foo"
+  }
+}
```

Comment:
can we tag it to make it easily understood by the user

Reply:
Tag as in how I didn't get it.

Reply:
I mean what is the old code, and what will be the new code kind of